src/libs/swapAndBridge/swapAndBridge.ts

```diff
@@ -84,15 +85,58 @@ const getActiveRoutesUpdateInterval = (minServiceTime?: number) => {
   return 15000
 }
 
-const buildSwapAndBridgeUserRequests = (
+const buildRevokeApprovalIfNeeded = async (
+  userTx: SocketAPISendTransactionRequest,
+  account: Account,
+  provider: RPCProvider
+): Promise<Call | undefined> => {
+  if (!userTx.approvalData) return
```

Comment:
This check here might not be correct in the case of basic accounts as the flow is written like this:

1. Check if we should do approve
2. Check if we should do revoke after checking if we should do approve
3. Do the swap

So, we first check `userTx.approvalData` and decide whether we should do an approval. Imagine the token is USDT on Ethereum and the user has already done an infinite approval prior to this request. We get `shouldApprove = false`.  
But then we enter `buildRevokeApprovalIfNeeded` which checks  `userTx.approvalData` which is true, so it continues. It tries to change the approval amount of USDT (which is currently infinite) and the function reverts. We mark `shouldApprove = true`

Imo, we should call `buildRevokeApprovalIfNeeded` only if `shouldApprove` is true as calculated before entering `buildRevokeApprovalIfNeeded`.